lib/galaxy/util/properties.py

```diff
@@ -85,15 +85,29 @@ def load_app_properties(
         properties.update(kwds)
 
     # update from env
-    override_prefix = "%sOVERRIDE_" % config_prefix
-    for key in os.environ:
-        if key.startswith(override_prefix):
-            config_key = key[len(override_prefix):].lower()
-            properties[config_key] = os.environ[key]
-        elif key.startswith(config_prefix):
-            config_key = key[len(config_prefix):].lower()
-            if config_key not in properties:
-                properties[config_key] = os.environ[key]
+    override_prefix = "override_"
+    yaml_prefix = "yaml_"
+    for key, val in os.environ.items():
+        override = False
+        if key.startswith(config_prefix):
+            # munch config_prefix
+            key = key[len(config_prefix):].lower()
```

Comment:
I would suggest not modifying the argument. We might need it, we might accidentally assume it's unchanged and use it as such somewhere else in the function, etc.)

Reply:
key only exists within the loop and sets things up for use later in the loop, the original is not valid after this point as all config variables are lowercase. This deduplicates converting to a valid variable identifier to just this line.

Reply:
I think I see what you meant. Changed this out.